core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -58,7 +58,7 @@ oppia.factory('UrlService', ['$window', function($window) {
     getTopicNameFromLearnerUrl: function() {
       var pathname = this.getPathname();
       if (pathname.match(/\/(story|topic)/g)) {
-        return pathname.split('/')[2];
+        return decodeURIComponent(pathname.split('/')[2]);
```

Comment:
Probably should add karma tests to check this.

Reply:
Done, added a test that correctly translates space in title.